modules/op-about-whenexpression.adoc

```diff
@@ -0,0 +1,174 @@
+// This module is included in the following assembly:
+//
+// *openshift_pipelines/understanding-openshift-pipelines.adoc
+
+[id="about-whenexpression_{context}"]
+= When expression
+
+`when` expressions help to guard task execution by setting a criteria for the execution of tasks within a Pipeline. `when` expressions contain a list of components, which allow a Task to run only when certain criteria are met.
+
+The key components of a `when` expression are as follows:
+
+* `input`: Represents static inputs or variables such as a parameter, task result, or execution status. A valid `input` must be specified. When not specified, the input value defaults to an empty string.
+* `operator`: Represents the relationship of an input to a set of `values`. A valid `operator` must be specified, which can be either `in` or `notin`.
+* `values`: Represents an array of string values. A non-empty array must be specified that can contain static values or variables such as parameters, results, or a bound state of a Workspace.
+
+The declared `when` expressions are evaluated before the Task is run. If all the `when` expressions evaluate to `True`, then the task is run. If any of the `when` expressions evaluates to `False`, the task is skipped.
+
+The *Pipeline Run details* page of the {product-title} web console shows the status of the tasks and the `when` expressions as follows:
+
+* All the criteria are met: Tasks and the `when` expression symbol, which is represented by a diamond shape, are in green.
+* Any of the criteria is not met: Task is skipped; skipped task and the `when` expression symbol are in grey.
+* None of the criteria is met: Task is skipped; skipped task and the `when` expression symbol are in grey.
+* Task run fails: Failed tasks and the `when` expression symbol are in red.
+
+With the `when` expressions, you can check a lot of scenarios. For example, whether:
+
+* The result of a previous Task is as expected.
+* A file in a Git repository has changed in the previous commits.
+* An image exists in the registry.
+* An optional Workspace has been provided.
+
+[NOTE]
+====
+* The parameters are read from the Pipeline and the `results` are read directly from the previous Tasks.
+* Using results in a `when` expression in a guarded Task introduces a resource dependency on the previous Task that produced the result.
+* `when` expressions are supported in the final tasks, which are specified using the `finally` field in the pipeline YAML file. `when` expressions in the final task provide efficient guarded execution and improved reusability of tasks.
+====
+
+The following example shows the `when` expressions for a PipelineRun to execute the `create-file` task only if the succeeding criteria is met: the `path` parameter is `README.md`,  and the `echo-file-exists` task executed only if the `exists` result from the `check-file` task is `yes`.
```

Comment:
This sentence is very confusing, suggest breaking it into two or three sentences: The following example shows the `when` expressions for a pipeline run. The pipeline run will execute the `create-file` task only if the succeeding criteria is met, that is, the value for the`path` parameter is `README.md`. It will execute the `echo-file-exists` task only if the `exists` result from the `check-file` task is `yes`.

One second thought since you explain all the when expressions in the call outs the last two sentences can be skipped and you can simply say: The following example shows the `when` expressions for a pipeline run.

Reply:
I think we should drop the word `succeeding`  from "succeeding criteria is met", because we can also specify negative assertions and its valid too. eg: If `path` `notin` `src/` then dont execute the task.

Reply:
In this example, the when expression for a PipelineRun executes the 'create-file' task when the following  criteria are met:
<list the criteria>